src/cascadia/TerminalControl/ControlCore.h

```diff
@@ -0,0 +1,243 @@
+        TYPED_EVENT(ScrollPositionChanged,     IInspectable, Control::ScrollPositionChangedArgs);
+        TYPED_EVENT(CursorPositionChanged,     IInspectable, IInspectable);
+        TYPED_EVENT(TaskbarProgressChanged,    IInspectable, IInspectable);
+        TYPED_EVENT(ConnectionStateChanged,    IInspectable, IInspectable);
+        TYPED_EVENT(HoveredHyperlinkChanged,   IInspectable, IInspectable);
+        TYPED_EVENT(RendererEnteredErrorState, IInspectable, IInspectable);
+        TYPED_EVENT(SwapChainChanged,          IInspectable, IInspectable);
+        TYPED_EVENT(RendererWarning,           IInspectable, Control::RendererWarningArgs);
+        TYPED_EVENT(RaiseNotice,               IInspectable, Control::NoticeEventArgs);
+        TYPED_EVENT(TransparencyChanged,       IInspectable, Control::TransparencyChangedEventArgs);
+        TYPED_EVENT(ReceivedOutput,            IInspectable, IInspectable);
+        // clang-format on
+
+    private:
+        bool _initializedTerminal{ false };
+        std::atomic<bool> _closing{ false };
+
+        TerminalConnection::ITerminalConnection _connection{ nullptr };
+        event_token _connectionOutputEventToken;
+        TerminalConnection::ITerminalConnection::StateChanged_revoker _connectionStateChangedRevoker;
+
+        std::unique_ptr<::Microsoft::Terminal::Core::Terminal> _terminal{ nullptr };
+
+        std::unique_ptr<::Microsoft::Console::Render::Renderer> _renderer{ nullptr };
+        std::unique_ptr<::Microsoft::Console::Render::DxEngine> _renderEngine{ nullptr };
+
+        IControlSettings _settings{ nullptr };
+
+        FontInfoDesired _desiredFont;
+        FontInfo _actualFont;
+
+        // storage location for the leading surrogate of a utf-16 surrogate pair
+        std::optional<wchar_t> _leadingSurrogate{ std::nullopt };
+
+        std::optional<COORD> _lastHoveredCell{ std::nullopt };
+        // Track the last hyperlink ID we hovered over
+        uint16_t _lastHoveredId{ 0 };
+
+        bool _isReadOnly{ false };
+
+        std::optional<interval_tree::IntervalTree<til::point, size_t>::interval> _lastHoveredInterval{ std::nullopt };
+
+        // These members represent the size of the surface that we should be
+        // rendering to.
+        double _panelWidth{ 0 };
+        double _panelHeight{ 0 };
+        double _compositionScaleX{ 0 };
+        double _compositionScaleY{ 0 };
+
+        winrt::fire_and_forget _asyncCloseConnection();
+
+        void _setFontSize(int fontSize);
+        void _updateFont(const bool initialUpdate = false);
+        void _refreshSizeUnderLock();
+        void _doResizeUnderLock(const double newWidth,
+                                const double newHeight);
+
+        void _sendInputToConnection(const winrt::hstring& wstr);
+        void _sendInputToConnection(std::wstring_view wstr);
+
+#pragma region TerminalCoreCallbacks
+        void _terminalCopyToClipboard(const std::wstring_view& wstr);

```

Comment:
no value in using a reference to a string view (which is just a pointer+size pair)